doc/fwutil/fwutil.md

```diff
@@ -462,6 +519,321 @@ Aborted!
+CPLD firmware update is going to be done with this command and the CPLD completion activity will be performed during cold reboot
+once all reboot processes are finished. The CPLD completion activity for this case is a power cycle triggered by the hw register setting.
+```
+admin@sonic:~# sudo fwutil auto_update fw --fw_image=fwpackage.tar.gz --boot=cold
+Firmware auto-update for boot_type cold is allowed
+MSN2700/CPLD1 firmware auto-update starting: /tmp/firmwareupdate/fwpackage/fwpackage/sn2700_cpld.mpfa with boot_type cold
+...
+CPLD1 firmware auto-update status: True with installed
+All firmware auto-update has been performed.
+```
+
+**Supported options:**
+3. -i|--image - update FW using current/next SONiC image
+4. -b|--boot - following boot option after the upgrade
+6. -z|--fw-image - firmware package downloaded during run time (this is an exclusive option from --image)
+
+**Note:** the default option is _--image=current_ and _--boot=any_
+
+##### 2.2.2.4.3 Platform Component Firmware Update Utility
+When Vendor doesn't have platform API ready to support all platform component APIs including the auto-update interface,
+the platform component utility will perform the equivalent process of `auto-update` platform component api.
+The component utility can perform the firmware update if the firmware update doesn't need any boot action required after the update.
+Otherwise, it will create a task file if any process or handling for the component firmware update needs to be done during the reboot
+and also if the update can be done for the specified reboot type.
+The componenet utility should be defined with key value `utility` in the component object of `platform_components.json` to be called by fwutil instead of the platform api.
+The task file will be platform-specific.
+
+**Example:**
+1. `platform_components.json` with platform specific utility.
+```json
+{
+    "chassis": {
+        "Chassis1": {
+            "component": {
+                "BIOS": {
+                    "firmware": "bios.bin",
+                    "version": "0ACLH003_02.02.010",
+                },
+                "CPLD": {
+                    "firmware": "cpld.bin",
+                    "utility": "cpldupdate",
+                    "version": "10",
+                },
+                "SSD": {
+                    "firmware": "SSD.bin",
+                    "utility": "ssd_fw_update",
+                    "version": "5",
+                }
+            }
+        }
+    }
+}
+```
+
+##### 2.2.2.4.3.1 Platform Component Firmware Update Utility Interface Requirement
+
+The Utility should support the minimum requirements to perform the fwutil auto-update interface. 
+The minimum requirement is that the component api's get_frimware_version, get_firmware_update_notification(), and auto_update_firmware() needs to be supported by the utility.
+Here are the interface requirements to support them.
+1. {utility} -s(--status) : able to retrieve the current firmware version:
```

Comment:
@sujinmkang how do we expect to use this info in FW tools when we don't have a way to read the original (before the update) firmware version?

Reply:
This api should retrieve the original version from the platform device. not the update firmware version.